packages/babel-core/src/transformation/file/options/option-manager.js

```diff
@@ -274,37 +274,24 @@ export default class OptionManager {
 
         // If the imported preset is a transpiled ES2015 module
         if (typeof val === "object" && val.__esModule) {
-          // Try to grab the default export.
           if (val.default) {
             val = val.default;
           } else {
-            // If there is no default export we treat all named exports as options
-            // and just remove the __esModule. This is to support presets that have been
-            // exporting named exports in the past, although we definitely want presets to
-            // only use the default export (with either an object or a function)
-            const { __esModule, ...rest } = val; // eslint-disable-line no-unused-vars
-            val = rest;
+            throw new Error("Preset must export a default export when using ES6 modules.");
           }
         }
 
-        // For compatibility with babel-core < 6.13.x, allow presets to export an object with a
-        // a 'buildPreset' function that will return the preset itself, while still exporting a
-        // simple object (rather than a function), for supporting old Babel versions.
-        if (typeof val === "object" && val.buildPreset) val = val.buildPreset;
-
-
-        if (typeof val !== "function" && options !== undefined) {
-          throw new Error(`Options ${JSON.stringify(options)} passed to ` +
-            (presetLoc || "a preset") + " which does not accept options.");
+        if (typeof val === "function") {
```

Comment:
If we can simply the logic (this code is the worst and so many hacks) then ok :+1: just want to make sure its easy to change it. We can write a migration guide + even try a codemod if that's useful.